packages/payload/src/collections/config/sanitize.ts

```diff
@@ -135,6 +136,12 @@ export const sanitizeCollection = async (
     }
 
     if (!sanitized.auth.disableLocalStrategy) {
+      const loginField = sanitized.auth.loginWithUsername ? 'username' : 'email'
+
+      if (loginField) {
+        authFields = authFields.concat(baseLoginField(loginField))
+      }
```

Comment:
This if statement is not needed, loginField will always be username or email. Was the intention to do something different?

Reply:
Nope not needed - removed!